src/components/CytoscapeGraph/Layout/GroupCompoundLayout.ts

```diff
@@ -34,6 +34,9 @@
     b. For every child set the relative position to its parent
  */
 
+import { CyNode } from '../CytoscapeGraphUtils';
```

Comment:
I haven't looked in depth, but wondering why this file had to change? I only remember using the concept of compound and not the group/app?

Reply:
Uh Oh, @josejulio is watching what I'm doing.  I would actually love for you to come back for a while and help with this stuff :)   The short story is that to help visualize multi-cluster stuff we are looking to have up to 3-levels of boxing.  So no longer can we assume that boxes are just for apps.  But now there could be up to 3 levels, where app boxes could have a namespace (or cluster) box as a parent, and a namespace box could have cluster box as a parent.  So "group-compound-layout" was basically assuming just app boxing, now it is starting to handle the nested boxing.

Reply:
> Uh Oh, @josejulio is watching what I'm doing

:eyes: 

> 3-levels of boxing

Ah, it makes sense, I never covered that case. I'll take a look at the file and leave comments. But I already forgot most of what is done here :)

Reply:
I'm not in love with the results, although not terrible, in general I don't understand why the layouts often want to "stack" the boxes as opposed to having a more horizontal result.  Actually, I'm not sure that's really the fault of our custom layout, but rather the "real" layouts.

Reply:
Could you show some screenshots of the results that you don't love?

Reply:
> I'm not in love with the results, although not terrible, in general I don't understand why the layouts often want to "stack" the boxes as opposed to having a more horizontal result. Actually, I'm not sure that's really the fault of our custom layout, but rather the "real" layouts.

I recall I did a stack-layout (or maybe because I used dagre?) somewhere in that same file so all the contents of the compounds are dagre-layouts. Maybe you can try to use the compound (if not already) in the inner boxes?